Form/Type/DynamicFormType.php

```diff
@@ -51,21 +79,34 @@ class DynamicFormType extends AbstractType
      * @param string $structureView
      * @param int $systemCollectionId
      * @param FormFieldTypePool $typePool
+     * @param CollectionTitleProviderPoolInterface $collectionTitlePool
+     * @param string $type
+     * @param stgring $typeId
+     * @param string $secret
      */
     public function __construct(
         Form $formEntity,
         $locale,
         $name,
         $structureView,
         $systemCollectionId,
-        FormFieldTypePool $typePool
+        FormFieldTypePool $typePool,
+        CollectionTitleProviderPoolInterface $collectionTitlePool,
+        $type,
+        $typeId,
+        $secret
```

Comment:
ignore the comment before!

$secret should be here the $checksum and build the checksum in the FormBuilder and check the checksum in the FormBuilder:buildByRequest (throw http bad request exception on false checksum! So the 400 error page is rendered)